CodeEdit/Features/Settings/Pages/SourceControlSettings/Models/IgnorePatternModel.swift

```diff
@@ -0,0 +1,283 @@
+    }
+
+    func loadPatterns() async {
+        await MainActor.run { loadingPatterns = true } // Ensure `loadingPatterns` is updated on the main thread
+
+        do {
+            let fileURL = try await gitIgnoreURL()
+            guard FileManager.default.fileExists(atPath: fileURL.path) else {
+                await MainActor.run {
+                    patterns = []
+                    loadingPatterns = false // Update on the main thread
+                }
+                return
+            }
+
+            if let content = try? String(contentsOf: fileURL) {
+                let parsedPatterns = content.split(separator: "\n")
+                    .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+                    .filter { !$0.isEmpty && !$0.starts(with: "#") }
+                    .map { GlobPattern(value: String($0)) }
+
+                await MainActor.run {
+                    patterns = parsedPatterns // Update `patterns` on the main thread
+                    loadingPatterns = false  // Ensure `loadingPatterns` is updated on the main thread
+                }
+            } else {
+                await MainActor.run {
+                    patterns = []
+                    loadingPatterns = false
+                }
+            }
+        } catch {
+            print("Error loading patterns: \(error)")
+            await MainActor.run {
+                patterns = []
+                loadingPatterns = false
+            }
+        }
+    }
+
+    func getPattern(for id: UUID) -> GlobPattern? {
+        return patterns.first(where: { $0.id == id })
+    }
+
+    func savePatterns() {
+        Task {
+            stopFileMonitor()
+            defer { Task { try? await startFileMonitor() } }
+
+            do {
+                let fileURL = try await gitIgnoreURL()
+                guard let fileContent = try? String(contentsOf: fileURL) else {
+                    writeAllPatterns()
+                    return
+                }
+
+                let lines = fileContent.split(separator: "\n", omittingEmptySubsequences: false).map(String.init)
+                let (patternToLineIndex, nonPatternLines) = mapLines(lines)
+                let globalCommentLines = extractGlobalComments(nonPatternLines, patternToLineIndex)
+
+                var reorderedLines = reorderPatterns(globalCommentLines, patternToLineIndex, nonPatternLines, lines)
+
+                // Ensure single blank line at the end
+                reorderedLines = cleanUpWhitespace(in: reorderedLines)
+
+                // Write the updated content back to the file
+                let updatedContent = reorderedLines.joined(separator: "\n")
+                try updatedContent.write(to: fileURL, atomically: true, encoding: .utf8)
+            } catch {
+                print("Error saving patterns: \(error)")
+            }
+        }
+    }
+
+    private func mapLines(_ lines: [String]) -> ([String: Int], [(line: String, index: Int)]) {
```

Comment:
Can we get a comment on these functions at the end of this file?

Reply:
We sure can